### PR TITLE
detect/transform: from_base64 transform

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #11299 

This PR provides the `from_base64` transform used to match on content decoded with base64 using the mode (default rfc4648).

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [6487](https://redmine.openinfosecfoundation.org/issues/6487)

Describe changes:
- New transform: `from_base64`
- Documentation
- Keyword parser in rust.

Updates:
- Doc: include what happens to the buffer if the input contains bytes that couldn't be decoded.
- Address other review comments.
- Format fix up

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1670

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPW2_autofp_suri_time.

ERROR: QA failed on SURI_TLPW1_suri_time.

ERROR: QA failed on SURI_TLPR1_suri_time.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW2_autofp_stats_chk | | |
| .uptime | 137 | 146 | 106.57% |
| | SURI_TLPR1_stats_chk | | |
| .uptime | 642 | 678 | 105.61% | 

Pipeline 21083

File: src/util-base64.c [hunk 1]
```diff
@@ -387,7 +387,7 @@ static int B64DecodeStringEndingSpaces(void)
     uint32_t consumed_bytes = 0, num_decoded = 0;
     uint8_t dst[10];
     Base64Ecode code = DecodeBase64(dst, sizeof(dst), (const uint8_t *)src, 9, &consumed_bytes,
-            &num_decoded, BASE64_MODE_RFC2045);
+            &num_decoded, Base64ModeRFC2045);
```
Comment (jasonish) [inline src/util-base64.c:390]: If you wanted to keep the all caps you can suppress the warning with `allow(non_camel_case_types)`.  Check the top of `core.rs` for an example.

But this is looking good.

**Review by catenacyber — CHANGES_REQUESTED** [review]
Could you review the ASAN SV report ?

Also maybe we want https://redmine.openinfosecfoundation.org/issues/7065 first

We do have 2 pure rust base64 decoders duplicates with the C ones...

**Comment by inashivb** [comment]
> Could you review the ASAN SV report ?
> 
> Also maybe we want https://redmine.openinfosecfoundation.org/issues/7065 first
> 
> We do have 2 pure rust base64 decoders duplicates with the C ones...

I'll work on this next week likely and will make the updates as needed to this keyword.
If this is ready, I think it does not need to be blocked.

**Comment by jlucovsky** [comment]
Continued in #11346

**Pull request closed without merging**